enhancements/installer/aws-custom-region-and-endpoints.md

```diff
@@ -0,0 +1,321 @@
+    // There must be only one CustomEndpoint for a service.
+    // +optional
+    APIEndpoints []CustomEndpoint `json:"customRegionOverride,omitempty"`
+    ...
+}
+```
+
+#### Destroy Cluster
+
+The `metadata.json` needs to store the custom endpoints from the install-config.yaml so that the users can delete the clusters without any previous state.
+
+#### AWS API access for assets
+
+Various assets in the installer use the caches session from the InstallConfig asset, this will be updated to provide a [resolver][aws-sdk-endpoints-resolver] like Kubernetes AWS [cloud-provider][k8s-aws-cloud-provider-get-resolver] so that we use the AWS SDK's default resolver for AWS APIs and override with the user provided ones when available.
+
+#### Configuring the Terraform AWS Provider
+
+The Terraform AWS provider doesn't accept random AWS API endpoint [overrides][terraform-aws-provider-api-endpoints], rather endpoint for each service needs to be explicitly specified. So the `TFVars` can pass a map object and the terraform `main.tf` can set each required endpoint override when certain key is present and set in the provide map object.
+
+#### Handling AMI
+
+##### Copying AMI to the region
+
+Terraform AWS Provider has a resource [`aws_ami_copy`][terraform-aws-provider-ami-copy] that can copy AMI from one region to the current region setup for the provider. The installer will copy the AMI from `us-east-1` region while continuing to encrypt it with the default KMS key.
+
+##### User provided AMI
+
+Already implemented.
+
+#### Validations
+
+Currently the installer only allows the users to specify region that have RHCOS AMIs published by maintaining a list of hard-coded regions. The installer will allow users to provide **any** region when using the `install-config.yaml` to create a cluster, but the terminal prompts should continue to use verified/hard coded list to display the regions.
+
+#### Default instances for custom regions
+
+AWS now allows users to find all the available instance type based on some constraints using [DescribeInstanceTypeOfferings][aws-describe-instance-type-offerings]. The installer should,
+
+1. Use this API to find m4/m5 instance types for the region that match the default CPU and memory requirements for control-plane and compute.
+
+2. Error out requiring the users to explicitly specify the desired instance types in cases where (1) doesn't return successful results.
+
+#### AWS API endpoints for cluster operators
+
+Since almost of the cluster operators that communicate to the the AWS API need the use the API endpoints provided by the user, the `Infrastructure` global configuration seems like the best place to store and discover this information.
+
+And in contrast to the current configuration in the `Infrastructure` object like `region` is not editable as day-2 operation, the users should be able to edit the AWS endpoints day-2, therefore this configuration best belongs in the `spec` section.
```

Comment:
cc @derekwaynecarr @deads2k @csrwng 

Because we had pushback previously from adding configuration to global configuration.

Reply:
@abhinavdahiya sgtm from a hosted deployment point of view